src/operations/AuthOperations.ts

```diff
@@ -0,0 +1,192 @@
+   * @param organizationId - Organization ID to switch to (required)
+   * @returns Auth result and encrypted session data
+   */
+  async switchOrganization(
+    session: Session,
+    organizationId: string,
+  ): Promise<{
+    auth: AuthResult;
+    encryptedSession: string;
+  }> {
+    // Delegate to refreshSession with explicit organization
+    return this.refreshSession(session, organizationId);
+  }
+
+  /**
+   * Refresh session operation.
+   *
+   * Calls WorkOS to refresh tokens (optionally switching organizations),
+   * encrypts the new session, and returns the auth result.
+   *
+   * @param session - Current session with refresh token
+   * @param organizationId - Optional organization ID to switch to during refresh
+   * @returns Auth result and encrypted session data
+   */
+  async refreshSession(
+    session: Session,
+    organizationId?: string,
+  ): Promise<{
+    auth: AuthResult;
+    encryptedSession: string;
+  }> {
+    // Determine which organization to use
+    let orgId = organizationId;
+    if (!orgId) {
+      // Extract org from current token (decodeJwt works even on expired tokens)
+      try {
+        const claims = this.core.parseTokenClaims<BaseTokenClaims>(
+          session.accessToken,
+        );
+        orgId = claims.org_id;
+      } catch {
+        // Token too malformed to parse - refresh without org context
+        // WorkOS will use whatever org is embedded in the refresh token
+      }
+    }
+
+    const result = await this.core.refreshTokens(session.refreshToken, orgId);
+
+    const newSession: Session = {
+      accessToken: result.accessToken,
+      refreshToken: result.refreshToken,
+      user: result.user,
+      impersonator: result.impersonator,
+    };
+
+    const encryptedSession = await this.core.encryptSession(newSession);
+
+    const claims = this.core.parseTokenClaims(result.accessToken);
+
+    const auth: AuthResult = {
+      user: result.user,
+      sessionId: claims.sid,
+      impersonator: result.impersonator,
+      accessToken: result.accessToken,
+      refreshToken: result.refreshToken,
+      claims,
+    };
+
+    return {
+      auth,
+      encryptedSession,
+    };
+  }
+
+  /**
+   * Get authorization URL for WorkOS authentication.
+   *
+   * Builds the WorkOS authorization URL with proper state encoding.
+   *
+   * @param options - Authorization URL options (returnPathname, screenHint, etc.)
+   * @returns The authorization URL
+   */
+  async getAuthorizationUrl(
+    options: GetAuthorizationUrlOptions = {},
+  ): Promise<string> {
+    return this.client.userManagement.getAuthorizationUrl({
+      provider: 'authkit',
+      redirectUri: options.redirectUri ?? this.config.getValue('redirectUri'),
+      screenHint: options.screenHint,
+      organizationId: options.organizationId,
+      loginHint: options.loginHint,
+      prompt: options.prompt,
+      clientId: this.config.getValue('clientId'),
+      state: options.returnPathname
```

Comment:
this is probably a weakness of the existing apis, but we're stopping our customers from other meaningful uses of state when we set it to returnPathname like this.

also this would be a good place for us to be preventing login-csrf (this sdk should be generating a nonce and passing it in here and storing in a cookie)

(login-csrf prevention could be a separate PR, but we shoul fix the issue of customer's not being able to set state now).